### PR TITLE
Header Enricher Docs: JavaConfig and DSL samples

### PR DESCRIPTION
**Pull request by garyrussell**
I just had to provide this example on SO, so I figured I'd just create a PR now...

### CODE REVIEW

File: src/reference/asciidoc/content-enrichment.adoc [hunk 1]
```diff
@@ -128,6 +128,52 @@ As you can see, by using SpEL for such simple cases, we no longer have to provid
...
+    return enricher;
+}
+
+@Bean
+@Transformer(inputChannel="enrichHeadersChannel", outputChannel="emailChannel")
+public HeaderEnricher enrichHeaders() {
+	Map<String, HeaderValueMessageProcessor<?>> headersToAdd = new HashMap<>();
+	headersToAdd.put("emailUrl", new StaticHeaderValueMessageProcessor<String>(this.imapUrl));
+	Expression expression = new SpelExpressionParser().parseExpression("payload.from[0].toString()");
+	headersToAdd.put("from", new ExpressionEvaluatingHeaderValueMessageProcessor<>(expression, String.class));
```
Comment (artembilan) [inline src/reference/asciidoc/content-enrichment.adoc]: Seems for me this is very long line for the PDF...

Comment (garyrussell) [inline src/reference/asciidoc/content-enrichment.adoc]: I see no errors from the build but you are right, it is folded to the next line with a bad indent.

Can you fix on merge?

Comment (artembilan) [inline src/reference/asciidoc/content-enrichment.adoc]: OK. No problem!
Just let me fix my XMPP work according to the Florian comments :smile:

Comment (garyrussell) [inline src/reference/asciidoc/content-enrichment.adoc]: No rush

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-integration/commit/0fd5f5ced885e0226751448458dd9b1f55b9a935

**Pull request closed without merging**